cylc/flow/platforms.py

```diff
@@ -58,10 +62,10 @@ def platform_from_name(platform_name=None, platforms=None):
         if re.fullmatch(platform_name_re, platform_name):
             platform_data = platforms[platform_name_re]
             if not platform_data:
-                platform_data = deepcopy(platforms['localhost'])
+                platform_data = platforms['localhost']
                 platform_data['remote hosts'] = platform_name
```

Comment:
Can just return the localhost platform unadulterated here?

Reply:
I need to do a wee bit more work here... This logic looks wrong to me,
I think that in it's current form this will only work if you have a platform name and nothing else. 
What it's not doing is filling the blanks in...